psychopy/experiment/components/sound/__init__.py

```diff
@@ -211,33 +211,35 @@ def writeFrameCode(self, buff):
         """
         # Write start code
         self.writeParamUpdates(buff, 'set every frame')
+
+        # write code for starting
         indented = self.writeStartTestCode(buff)
         if indented:
             if self.params['syncScreenRefresh'].val:
                 code = ("%(name)s.play(when=win)  # sync with win flip\n") % self.params
             else:
                 code = "%(name)s.play()  # start the sound (it finishes automatically)\n" % self.params
-            buff.writeIndented(code)
-        buff.setIndentLevel(-indented, relative=True)
-
-        # Write stop code
-        indented = self.writeStopTestCode(buff)
-        if indented:
-            code = ("%(name)s.stop()\n")
             buff.writeIndentedLines(code % self.params)
-        # because of the 'if' statement of the time test
         buff.setIndentLevel(-indented, relative=True)
 
-        # Update status
+        # update status according to playback
         code = (
             "# update %(name)s status according to whether it's playing\n"
             "if %(name)s.isPlaying:\n"
```

Comment:
Good point yeah, FINISHED is there because the Sound can finish playing without the Component finishing, but the same isn't true of starting so that line isn't needed.

I guess the concern is when people have Code components set up to call `.start()` and are using `.status == STARTED` to check whether it's still playing before their logic to stop it, as previously calling `.start()` would set `.status`, but in these instances I think we should be advising users to set `.status = STARTED` themselves, or query `.isPlaying` instead of `.status`.

Reply:
Agreed. If they are manually calling `.start()` but using `.status` to check, they should set `.status` manually as well as demonstrated by Builder generated scripts.